override invocation name in skill.conf

Yahoo-Mike opened the pull request:
This relates to issue #8 .

These changes allow user to override "Jellyfin Player" as invocation name in the interaction model.  It also allows user to replace the phrase "Jellyfin Player" in the manifest example phrases.

In the skills.conf there are two sections: [en-US] and [de-DE].  Two keys are allowed: _invocation_name_ and _display_name_.  The _invocation_name_ will override the default invocation name for that locale ("Jellyfin Player").  The _display_name_ will replace the phrase "Jellyfin Player" in the three example phrases for that locale.

The changes to main.py are suggestions.  I have tested the changes I've made and they work.  I am not a python programmer, so there may be more elegant ways to implement the code.  Feel free to suggest improvements.

infinityofspace commented:
Thank you for the PR. Good work!

There are just a few minor adjustments that we can discuss.

Yahoo-Mike commented:
I think the latest commit addresses all the issues you raised.  I moved the code that changes SKILL_MANIFEST into manifest.py.

I also found a bug in what I had before: when a user removed a local override from skill.conf, the old override remained on Alexa.  This commit will reload the default invocation name and example phrases, then (if they are different from what's currently on Alexa) it will force a rebuild.

infinityofspace commented:
Looks very good now.
Thanks for your efforts to improve the skill.
